### ISSUE TITLE
Queueing Spotify track pauses playback and breaks queue

### ISSUE THREAD

**Issue by n1mda** [bencevans/node-sonos#118]
Using `addSpotify(track_id, callback)` the track get set as the current track, but playback is paused, and it  Sonos no longer uses the queue
##

**Comment by sseiber** [comment]
addSpotify is returning:
HTTP response code 500 for "urn:schemas-upnp-org:service:AVTransport:1#SetAVTransportURI"

I'm using "spotify:track:1sTAjaPsXRBfdSLrC0PVUS" as the track id and I have tried others.

I find that it also stops current playback, but it does not set the current track, or pause playback.

I also attempted to add a Spotify track id with the sonos-cli tool, as well as the sonos_cli tool and the node-sonos-http-api. Could it be that the SpotifyAPI has changed?

**Comment by bencevans** [comment]
I don't believe this is broken as per @sseiber's comment but do believe it's poor in design.

`.addSpotify(trackId, callback)` just accepts the track id, NOT the full URI. For example with your Spotify URI 'spotify:track:1sTAjaPsXRBfdSLrC0PVUS', you need to use `.addSpotify('1sTAjaPsXRBfdSLrC0PVUS', callback)` to set the track.

I've found the same as @n1mda when just a Spotify track id is provided, the track is set but playback paused.  

Ideally I'd like `.play(uri, callback)`, `.queue(uri, callback)`, `.queueNext(uri, callback)` to accept and correctly handle Spotify URIs and place in the appropriate place in queue and drop `addSpotify`.

**Comment by sseiber** [comment]
Thanks for the reply Ben.

I've changed the way I'm getting the Sonos object and I'm now observing the playback stops as you and @n1mda describe, but I'm still getting the error:
HTTP response code 500 for "urn:schemas-upnp-org:service:AVTransport:1#SetAVTransportURI"
and the queue is disconnected.

My code is this:

const Sonos = require('sonos');
var sonos = new Sonos.Sonos(sonosIp);
sonos.addSpotify('1sTAjaPsXRBfdSLrC0PVUS', (err, result) => {
    if (err) {
        return next(err);
    }
    return next(null, result);
});

I've looked through the source and it looks like is really just setting up the metadata and uri for a call to queuenext. Has that format changed somehow? Let me know if you have any suggestions on things I could try. E.g. turn on the rand calculation, changes to the metadata, etc.

**Comment by sseiber** [comment]
I did a bit of investigating today with WireShark. I found that when I use the Sonos app to add a Spotify track to the queue I see a SOAP body with AddURIToQueue in the body. When I use the addSpotify api from the library I see a SOAP body with SetAVTransportURI in the body.

I experimented with this and modified the addSpotify api to call Sonos.queue (which uses the AddURIToQueue wrapper) vs. the Sonos.queueNext (which uses the SetAVTransportURI wrapper).  This actually caused a Spotify item to be created in the queue; however it didn't render with the name and album art. It just created a generic queue entry with the Spotify URI. It didn't play and was not playable by clicking on it. But, it also didn't stop the currently playing song.

I'll continue to experiment.
